lib/models/addon.js

```diff
@@ -221,7 +221,7 @@ let addonProto = {
     p.setupRegistry(this);
 
     if (!this.name) {
-      throw new SilentError('An addon must define a `name` property.');
+      throw new SilentError(`An addon must define a \`name\` property (found at ${this.root}).`);
```

Comment:
yeah, although that should probably be a different refactoring. but since we're planning such a fallback, testing this PR here might not actually make as much sense. 🤷‍♂️